webapp/resources/messages.properties

```diff
@@ -443,3 +443,75 @@ cas.mfa.registerdevice.pagetitle=Register Device
 cas.mfa.registerdevice.label.name=Name
 cas.mfa.registerdevice.button.register=Register
 
+cas.pac4j.twitterclient.name=Twitter
+cas.pac4j.twitterclient.class=btn-twitter
+cas.pac4j.foursquareclient.name=Foursquare
+cas.pac4j.foursquareclient.class=btn-foursquare
+cas.pac4j.windowsliveclient.name=Windows Live
+cas.pac4j.windowsliveclient.class=btn-windowslive
+cas.pac4j.yahooclient.name=Yahoo
+cas.pac4j.yahooclient.class=btn-Yahoo
+cas.pac4j.bitbucketclient.name=Bitbucket
+cas.pac4j.bitbucketclient.class=btn-bitbucket
+cas.pac4j.wordpressclient.name=WordPress
+cas.pac4j.wordpressclient.class=btn-wordpress
+cas.pac4j.google2client.name=Google
+cas.pac4j.google2client.class=btn-google
+cas.pac4j.githubclient.name=Github
+cas.pac4j.githubclient.class=btn-github
+cas.pac4j.dropboxclient.name=Dropbox
+cas.pac4j.dropboxclient.class=btn-dropbox
+cas.pac4j.linkedinclient.name=LinkedIn
+cas.pac4j.linkedinclient.class=btn-linkedin
+cas.pac4j.facebookclient.name=Facebook
+cas.pac4j.facebookclient.class=btn-facebook
+cas.pac4j.paypalclient.name=PayPal
+cas.pac4j.paypalclient.class=btn-paypal
+cas.pac4j.casclient.name=CAS
+cas.pac4j.casclient.class=btn-cas
+cas.pac4j.casclient1.name=CAS 2
+cas.pac4j.casclient1.class=btn-cas
+cas.pac4j.casclient2.name=CAS 3
+cas.pac4j.casclient2.class=btn-cas
+cas.pac4j.casclient3.name=CAS 4
+cas.pac4j.casclient3.class=btn-cas
+cas.pac4j.saml2client.name=SAML2
+cas.pac4j.saml2client.class=btn-saml2
+cas.pac4j.saml2client1.name=SAML2 1
+cas.pac4j.saml2client1.class=btn-saml2
+cas.pac4j.saml2client2.name=SAML2 2
+cas.pac4j.saml2client2.class=btn-saml2
+cas.pac4j.saml2client3.name=SAML2 3
+cas.pac4j.saml2client3.class=btn-saml2
+cas.pac4j.genericoauth20client.name=Generic OAuth 2.0
+cas.pac4j.genericoauth20client.class=btn-oauth20
+cas.pac4j.genericoauth20client1.name=Generic OAuth 2.0 2
+cas.pac4j.genericoauth20client1.class=btn-oauth20
+cas.pac4j.genericoauth20client2.name=Generic OAuth 2.0 3
+cas.pac4j.genericoauth20client2.class=btn-oauth20
+cas.pac4j.genericoauth20client3.name=Generic OAuth 2.0 4
```

Comment:
All great. This bit here I think needs to be looked at more. Two things specifically:

1. I don't think CSS classes are things that belong in a `messages.properties` file. That's a file used a language bundle and css classes don't belong there. (There is no need to translate a CSS class for example).
2. Probably need to find a better strategy than `cas.pac4j.saml2client3.name=SAML2 3`. The more automatic you make this better....and while this might be far fetched, there shouldn't be a limitation on how many clients of SAML or CAS are supported here. (Today, I am forced to add keys for the 4th one, and I would like to avoid that technical limitation).

My suggestion would be to remove all such keys from the properties file here, and move them to the relevant html page. I can promise you that for a real deployment, all of the html/UI is going to get ripped out and replaced with something entirely custom and in those cases, I would like the entirety of UI changes to be removed from the deployment if needed, rather than replacing the HTML and leaving the properties.

Makes sense?